src/rebar_file_utils.erl

```diff
@@ -273,6 +275,22 @@ canonical_path([_|Acc], [".."|Rest])  -> canonical_path(Acc, Rest);
 canonical_path([], [".."|Rest])       -> canonical_path([], Rest);
 canonical_path(Acc, [Component|Rest]) -> canonical_path([Component|Acc], Rest).
 
+%% returns canonical target of path if path is a link, otherwise returns path
+-spec resolve_link(string()) -> string().
+
+resolve_link(Path) ->
+    case file:read_link(Path) of
+        {ok, Target} ->
+            canonical_path(filename:absname(Target, filename:dirname(Path)));
+        {error, _} -> Path
+    end.
```

Comment:
Yeah, I added the switch there to introduce the functionality as an option,
in case people are scraping output.

I personally would not just change this without a gradual introduction -
your output is an implicit contract whether it's documented or not.

One could argue the default should be 'relative' though. This would allow
someone to opt out in the short term and refactor as needed (e.g. if
backward compatibility is deprecated).

On Tue, May 3, 2016 at 5:16 AM, Fred Hebert <notifications@github.com>
wrote:

> In src/rebar_file_utils.erl
> <https://github.com/erlang/rebar3/pull/1174#discussion_r61871296>:
>
> > @@ -273,6 +275,22 @@ canonical_path([_|Acc], [".."|Rest])  -> canonical_path(Acc, Rest);
> >  canonical_path([], [".."|Rest])       -> canonical_path([], Rest);
> >  canonical_path(Acc, [Component|Rest]) -> canonical_path([Component|Acc], Rest).
> >
> > +%% returns canonical target of path if path is a link, otherwise returns path
> > +-spec resolve_link(string()) -> string().
> > +
> > +resolve_link(Path) ->
> > +    case file:read_link(Path) of
> > +        {ok, Target} ->
> > +            canonical_path(filename:absname(Target, filename:dirname(Path)));
> > +        {error, _} -> Path
> > +    end.
>
> I would be of the opinion not to configure it. Just display the most
> helpful path there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/erlang/rebar3/pull/1174/files/da0e2c22959c976420b60def760b348aa200d6e2#r61871296>
>